daemon/network.go

```diff
@@ -113,7 +114,12 @@ func (daemon *Daemon) CreateNetwork(name, driver string, ipam network.IPAM, opti
 	}
 
 	nwOptions = append(nwOptions, libnetwork.NetworkOptionIpam(ipam.Driver, "", v4Conf, v6Conf))
-	nwOptions = append(nwOptions, libnetwork.NetworkOptionDriverOpts(options))
+	nwOptions = append(nwOptions, libnetwork.NetworkOptionDriverOpts(driverOptions))
+	if len(networkOptions) > 0 {
+		if _, ok := networkOptions[netlabel.Internal]; ok {
+			nwOptions = append(nwOptions, libnetwork.NetworkOptionInternalNetwork())
```

Comment:
@calavera @icecrime thats correct. libnetwork doesn't mix these options. But we designed it as a variadic arguments to avoid changing the APIs for all the various options that we will support.
If we have to reflect that on the calling side, then am afraid we have to change the API.